app/pages/api/supply.ts

```diff
@@ -0,0 +1,111 @@
+import { getContract, getStaticProvider } from "@klimadao/lib/utils";
+import { formatUnits } from "ethers/lib/utils";
+import { DEFAULT_NETWORK } from "lib/constants";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export const config = {
+  api: {},
+  // Specifies the maximum allowed duration for this function to execute (in seconds)
+  maxDuration: 5,
+};
+
+type APIResponse =
+  | number
+  | {
+      totalSupply: number;
+      circulatingSupply: number;
+    }
+  | {
+      error: string;
+    };
+
+// This can be simplified to be just a string[]
+// but IERC20 ABI doesn't have a decimal exposed
+const DEFAULT_TOKENS = [
+  {
+    name: "klima",
+    decimals: 9,
+  },
+  {
+    name: "bct",
+    decimals: 18,
+  },
+];
+
+/**
+ * @description fetch total supply for a token
+ * @param req
+ * @param res
+ * @example GET /api/supply?token=klima Returns {APIResponse}
+ * @example GET /api/supply?type=total-supply&token=klima Returns number
```

Comment:
these examples are still using the older long `type` values with the `-supply` suffix - should be fixed to shorter names

Reply:
Believe it or not I didn't push the latest changes I made last night. Will do that as well.